includes/admin/class-wc-stripe-admin-notices.php

```diff
@@ -444,6 +449,61 @@ public function payment_methods_check_environment() {
 		}
 	}
 
+	/**
+	 * Adds a notice to the order details page if the subscription payment method has been detached.
+	 *
+	 * @return void
+	 */
+	public function subscription_check_detachment() {
+		if ( ! self::is_subscription_edit_page() ) {
+			return;
+		}
+
+		global $theorder;
+
+		// If $theorder is empty (i.e. non-HPOS), fallback to using the global post object.
+		if ( empty( $theorder ) && ! empty( $GLOBALS['post']->ID ) ) {
+			$theorder = wcs_get_subscription( $GLOBALS['post']->ID );
+		}
+
+		if ( ! isset( $theorder ) || ! $theorder instanceof WC_Subscription ) {
+			return;
+		}
+
+		$subscription = $theorder;
```

Comment:
Populating `$theorder` using local code feels risky, especially if other code/plugins have checks driven by `$theorder` being defined. I think we should stick to a local variable.
```suggestion
		global $theorder;

		$subscription = null;

		if ( isset( $theorder ) ) {
			$subscription = $theorder;
		}
		// If $theorder is empty (i.e. non-HPOS), fallback to using the global post object.
		elseif ( empty( $theorder ) && ! empty( $GLOBALS['post']->ID ) ) {
			$subscription = wcs_get_subscription( $GLOBALS['post']->ID );
		}

		if ( ! isset( $subscription ) || ! $subscription instanceof WC_Subscription ) {
			return;
		}

```